nibabel/cmdline/tests/test_convert.py

```diff
@@ -0,0 +1,152 @@
+#!python
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+
+import pytest
+
+import numpy as np
+
+import nibabel as nib
+from nibabel.testing import test_data
+from nibabel.cmdline import convert
+
+
+def test_convert_noop(tmp_path):
+    infile = test_data(fname='anatomical.nii')
+    outfile = tmp_path / 'output.nii.gz'
+
+    orig = nib.load(infile)
+    assert not outfile.exists()
+
+    convert.main([str(infile), str(outfile)])
+    assert outfile.is_file()
+
+    converted = nib.load(outfile)
+    assert np.allclose(converted.affine, orig.affine)
+    assert converted.shape == orig.shape
+    assert converted.get_data_dtype() == orig.get_data_dtype()
+
+    with pytest.raises(FileExistsError):
+        convert.main([str(infile), str(outfile)])
+
+    convert.main([str(infile), str(outfile), '--force'])
+    assert outfile.is_file()
```

Comment:
Not too important, but is there a way to make sure that this is the file that was created in the forced run, and not the one that existed there before? You might worry that the `force` flag would somehow make the program do nothing.